### PR TITLE
[distribution] Add support for global values

### PR DESCRIPTION
**Pull request by chukka**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
[distribution] Add support for global values

### CODE REVIEW

File: stable/distribution/templates/_helpers.tpl [hunk 1]
```diff
@@ -79,3 +79,83 @@ Scheme (http/https) based on Access TLS enabled/disabled
...
+{{/*
+Resolve imagePullSecrets value
+*/}}
+{{- define "distribution.imagePullSecrets" -}}
+{{- if .Values.global.imagePullSecrets }}
+imagePullSecrets:
+{{- range .Values.global.imagePullSecrets }}
+  - name: {{ . }}
+{{- end }}
+{{- else if .Values.imagePullSecrets }}
```
Comment (danielezer) [inline stable/distribution/templates/_helpers.tpl:120]: Shouldn’t the non-global value be first? I mean, doesn’t the specific values take precedence over the global ones?

File: stable/distribution/templates/distribution-statefulset.yaml [hunk 2]
```diff
@@ -56,29 +55,29 @@ spec:
...
-        {{- if or .Values.distribution.joinKey .Values.distribution.joinKeySecretName }}
+        {{- if or .Values.distribution.joinKey .Values.distribution.joinKeySecretName .Values.global.joinKey .Values.global.joinKeySecretName }}
         - name: DISTRIBUTION_JOIN_KEY
           valueFrom:
             secretKeyRef:
-              name: "{{ .Values.distribution.joinKeySecretName | default (include "distribution.fullname" .) }}"
+              name: {{ include "distribution.joinKeySecretName" . }}
               key: join-key
         {{- end }}
         {{- if or .Values.distribution.masterKey .Values.distribution.masterKeySecretName }}
```
Comment (danielezer) [inline stable/distribution/templates/distribution-statefulset.yaml:76]: You didn’t add the global values to this if

File: stable/distribution/values.yaml [hunk 3]
```diff
@@ -3,11 +3,75 @@
...
+  #      limits:
+  #        memory: "128Mi"
+  #        cpu: "100m"
+
 # Common
 initContainerImage: docker.bintray.io/alpine:3.12
 
 # For supporting pulling from private registries
-imagePullSecrets:
+# imagePullSecrets:
```
Comment (danielezer) [inline stable/distribution/values.yaml:73]: This is breaking. Say that in the changelog. Also, I don’t see a reason to comment out the imagePullSecrets. You can uncomment that and just comment out myRegistryKeySecretName

**Review by danielezer — CHANGES_REQUESTED** [review]

File: stable/distribution/templates/_helpers.tpl (hunk 1 above)
Comment (chukka) [reply stable/distribution/templates/_helpers.tpl:120]: No.  Global value from parent chart takes precedence over subchart value

File: stable/distribution/values.yaml (hunk 3 above)
Comment (chukka) [reply stable/distribution/values.yaml:73]: Already added

File: stable/distribution/Chart.yaml [hunk 4]
```diff
@@ -11,15 +11,6 @@ keywords:
...
-  name: amithins
-- email: daniele@jfrog.com
-  name: danielezer
-- email: eldada@jfrog.com
-  name: eldada
-- email: ramc@jfrog.com
-  name: chukka
-- email: rimasm@jfrog.com
-  name: rimusz
+- name: Jfrog
```
Comment (JfrogPrasanna) [inline stable/distribution/Chart.yaml:14]: Chat Maintainers at JFrog

File: stable/distribution/templates/distribution-statefulset.yaml (hunk 2 above)
Comment (chukka) [reply stable/distribution/templates/distribution-statefulset.yaml]: Fixed

File: stable/distribution/Chart.yaml (hunk 4 above)
Comment (chukka) [reply stable/distribution/Chart.yaml]: Fixed

**Review by danielezer — APPROVED** [review]

**Review by rahulsadanandan — APPROVED** [review]

**Review by rimusz — APPROVED** [review]
LGTM

**Pull request merged**